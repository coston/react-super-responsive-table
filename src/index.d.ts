declare module 'react-super-responsive-table' {
    export class Table extends React.Component<TableProps & any, any> {
    }
  
    interface TableProps{
      className?: string,
    }
  
    export class Tbody extends React.Component<TbodyProps & any> {
    }
  
    interface TbodyProps{
    }
  
    export class Td extends React.Component<TdProps & any> {
    }
  
    interface TdProps{
    }
  
    export class Th extends React.Component<ThProps & any>{
    }
  
    interface ThProps{ 
    }
  
    export class Thead extends React.Component<TheadProps & any>{
    }
  
    interface TheadProps{
      children: any
    }
  
    export class Tr extends React.Component<TrProps & any>{
    }
  
    interface TrProps{
    }
  }