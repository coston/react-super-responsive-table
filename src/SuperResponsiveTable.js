import React from 'react'
import { Provider, Consumer } from './tableContext'

const omit = (obj, omitProps) =>
  Object.keys(obj)
    .filter(key => omitProps.indexOf(key) === -1)
    .reduce((returnObj, key) => ({ ...returnObj, [key]: obj[key] }), {})

const allowed = props => omit(props, ['inHeader', 'columnKey', 'headers'])

export class Table extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      headers: {},
    }
  }
  render() {
    const { headers } = this.state
    const classes = (this.props.className || '') + ' responsiveTable'
    return (
      <Provider value={headers}>
        <table {...allowed(this.props)} className={classes} />
      </Provider>
    )
  }
}

export const Thead = props => (
  <thead {...allowed(props)}>
    {React.cloneElement(props.children, { inHeader: true })}
  </thead>
)

class TrInner extends React.Component {
  constructor(props) {
    super(props)
    const { headers } = props
    if (headers && props.inHeader) {
      React.Children.map(props.children, (child, i) => {
        if (child) {
          headers[i] = child.props.children
        }
      })
    }
  }
  render() {
    const { children } = this.props
    return (
      <tr {...allowed(this.props)}>
        {children &&
          React.Children.map(
            children,
            (child, i) =>
              child &&
              React.cloneElement(child, {
                key: i,
                columnKey: i,
              })
          )}
      </tr>
    )
  }
}

export const Tr = props => (
  <Consumer>{headers => <TrInner {...props} headers={headers} />}</Consumer>
)

export const Th = props => <th {...allowed(props)} />
export const Tbody = props => <tbody {...allowed(props)} />

class TdInner extends React.Component {
  render() {
    if (this.props.colSpan) {
      return <td {...allowed(this.props)} />
    }
    const { headers, children, columnKey } = this.props
    const classes = (this.props.className || '') + ' pivoted'
    return (
      <td {...allowed(this.props)} className={classes}>
        <div className="tdBefore">{headers[columnKey]}</div>
        {children || <div>&nbsp;</div>}
      </td>
    )
  }
}

export const Td = props => (
  <Consumer>{headers => <TdInner {...props} headers={headers} />}</Consumer>
)
