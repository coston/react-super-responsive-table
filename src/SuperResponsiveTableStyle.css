/* inspired by: https://css-tricks.com/responsive-data-tables/ */
.responsiveTable {
  width: 100%;
}

.responsiveTable td .tdBefore {
  display: none;
}

@media screen and (max-width: 40em) {
  /*
    Force table elements to not behave like tables anymore
    Hide table headers (but not display: none;, for accessibility)
  */

  .responsiveTable table,
  .responsiveTable thead,
  .responsiveTable tbody,
  .responsiveTable th,
  .responsiveTable td,
  .responsiveTable tr {
    display: block;
  }

  .responsiveTable thead tr {
    position: absolute;
    top: -9999px;
    left: -9999px;
    border-bottom: 2px solid #333;
  }

  .responsiveTable tbody tr {
    border: 1px solid #000;
    padding: .25em;
  }

  .responsiveTable td.pivoted {
    /* Behave like a "row" */
    border: none !important;
    display: block;
    position: relative;
    text-align: left !important;
    white-space: pre-wrap;
    overflow-wrap: break-word;
  }

  .responsiveTable td .tdBefore {
    /* Now like a table header */
    display: inline-flex;

    /* Top/left values mimic padding */
    left: 1rem;
    width: calc(50% - 20px);
    white-space: pre-wrap;
    overflow-wrap: break-word;
    text-align: left !important;
    font-weight: 600;
  }

  .responsiveTable td .tdAfter {
    display: inline-flex;
    padding-left: 10px;
  }
}
