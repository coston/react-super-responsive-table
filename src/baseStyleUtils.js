const css = (strings, ...expressions) => {
  let str = ''
  strings.forEach((string, i) => {
    str += string + (expressions[i] || '')
  })

  return str
}

const generateBaseStyles = ({ breakpoint = '40em' }) => css`
.super-responsive-table__th_pivoted {
  display: none;
}

@media screen and (max-width: ${breakpoint}) {
  .super-responsive-table__table,
  .super-responsive-table__thead,
  .super-responsive-table__tbody,
  .super-responsive-table__th,
  .super-responsive-table__td,
  .super-responsive-table__tr {
    display: block;
  }

  .super-responsive-table__thead {
    position: absolute;
    top: -9999px;
    left: -9999px;
  }

  .super-responsive-table__td_pivoted {
    position: relative;
    margin-left: 50%;
  }

  .super-responsive-table__th_pivoted {
    box-sizing: border-box;
    position: absolute;
    top: 0;
    left: -100%;
    width: 100%;
    height: 100%;
  }
}
`

const getStyleTag = () => {
  const existingStyleTag = document.head.querySelector(
    'style[data-super-responsive-table]'
  )

  if (existingStyleTag) {
    return existingStyleTag
  }

  const newStyleTag = document.createElement('style')
  newStyleTag.setAttribute('data-super-responsive-table', '')
  document.head.appendChild(newStyleTag)

  return newStyleTag
}

export const injectBaseStyles = props => {
  const styleTag = getStyleTag()
  styleTag.innerHTML = generateBaseStyles(props)
}
