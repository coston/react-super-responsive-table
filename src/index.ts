import Table from './components/Table';
import Tbody from './components/Tbody';
import Td from './components/Td';
import Th from './components/Th';
import Thead from './components/Thead';
import Tr from './components/Tr';

export { Table, Tbody, Td, Th, Thead, Tr };
