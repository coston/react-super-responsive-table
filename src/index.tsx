import React from 'react'

import Table from './components/Table';
import Tbody from './components/Tbody';
import Td from './components/Td';
import Th from './components/Th';
import Thead from './components/Thead';
import Tr from './components/Tr';


const Ham = () => {
    console.log("OOOKDSFSDF")
return (<p>TEST ok</p>)
};


export { Ham, Table, Tbody, Td, Th, Thead, Tr };
