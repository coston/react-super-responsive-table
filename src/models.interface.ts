
export interface TableSection extends React.HTMLProps<HTMLTableSectionElement>{
    children: any;
    inHeader: boolean;
  }

  export interface TableRow extends React.HTMLProps<HTMLTableRowElement>{
    children: any;
    inHeader: boolean;
    setHeaders: any;
  }