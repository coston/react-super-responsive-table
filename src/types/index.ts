import { Dispatch, ReactNode, SetStateAction } from 'react';

export type OmitProps = {
  inHeader?: boolean;
  columnKey?: number;
  headers?: string[];
  forwardedRef?: React.Ref<unknown>;
  className?: string;
  colSpan?: number;
  children?: ReactNode;
};

export type HeaderContextType = {
  headers: string[];
  setHeaders: Dispatch<SetStateAction<string[]>>;
};

export type TdProps = OmitProps & {
  columnKey?: number;
};
