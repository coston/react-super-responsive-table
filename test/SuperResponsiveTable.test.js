import React from 'react'
import { Table, Thead, Tbody, Tr, Th, Td } from '../src/SuperResponsiveTable'
import renderer from 'react-test-renderer'

test('Render Table', () => {
  const wrapper = shallow(
    <Table>
      <Thead>
        <Tr>
          <Th>Annual Conference</Th>
          <Th>Year</Th>
          <Th>Location</Th>
          <Th>President</Th>
          <Th>Program Chair</Th>
          <Th>Conference Theme</Th>
        </Tr>
      </Thead>
      <Tbody>
        <Tr>
          <Td>31</Td>
          <Td>2017</Td>
          <Td>Alabama Community College System (ACCS)</Td>
          <Td>Mr. Toner Evans, Samford University</Td>
          <Td>Ms. Kelly Birchfield, Auburn University Montgomery</Td>
        </Tr>
      </Tbody>
    </Table>
  )
  expect(wrapper).toMatchSnapshot()
})


test('Render table with an only one column', () => {
  const wrapper = shallow(
    <Table>
      <Thead>
      <Tr>
        <Th>Annual Conference</Th>
      </Tr>
      </Thead>
      <Tbody>
      <Tr>
        <Td>31</Td>
      </Tr>
      </Tbody>
    </Table>
  )
  expect(wrapper).toMatchSnapshot()
})

test('Render table without any column', () => {
  const wrapper = shallow(
    <Table>
      <Thead>
      <Tr>
      </Tr>
      </Thead>
      <Tbody>
      <Tr>
      </Tr>
      </Tbody>
    </Table>
  )
  expect(wrapper).toMatchSnapshot()
})

test('Render table with only header and empty row', () => {
  let component = renderer.create(
    <Table>
      <Thead>
      <Tr>
        <Td>Test Column</Td>
      </Tr>
      </Thead>
      <Tbody>
      <Tr>
      </Tr>
      </Tbody>
    </Table>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Render table with conditional column', () => {
  let component = renderer.create(
    <Table>
      <Thead>
      <Tr>
        {false && <Td>Test Header Column</Td>}
      </Tr>
      </Thead>
      <Tbody>
      <Tr>
        {false && <Td>Test Body Column</Td>}
      </Tr>
      </Tbody>
    </Table>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Render table with conditional and unconditional column', () => {
  let component = renderer.create(
    <Table>
      <Thead>
      <Tr>
        <Td>C1</Td>
        {false && <Td>C2</Td>}
        <Td>C3</Td>
      </Tr>
      </Thead>
      <Tbody>
      <Tr>
        <Td>V1</Td>
        {false && <Td>V2</Td>}
        <Td>V3</Td>
      </Tr>
      </Tbody>
    </Table>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Render table with more columns in body', () => {
  let component = renderer.create(
    <Table>
      <Thead>
      <Tr>
        <Td>C1</Td>
        {false && <Td>C2</Td>}
        <Td>C3</Td>
      </Tr>
      </Thead>
      <Tbody>
      <Tr>
        <Td>V1</Td>
        {false && <Td>V2</Td>}
        <Td>V3</Td>
        <Td>V4</Td>
        <Td>V6</Td>
      </Tr>
      </Tbody>
    </Table>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Render table with more columns in header', () => {
  let component = renderer.create(
    <Table>
      <Thead>
      <Tr>
        <Td>C1</Td>
        {false && <Td>C2</Td>}
        <Td>C3</Td>
        <Td>C4</Td>
        <Td>C5</Td>
        <Td>C6</Td>
      </Tr>
      </Thead>
      <Tbody>
      <Tr>
        <Td>V1</Td>
        {false && <Td>V2</Td>}
        <Td>V3</Td>
      </Tr>
      </Tbody>
    </Table>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Table updates correctly upon change of key prop', () => {
  // stateful container above
  const Parent = ({ headers, i }) => (
    <Table key={i}>
      <Thead>
      <Tr>
        <Th>{headers[0]}</Th>
        <Th>{headers[1]}</Th>
      </Tr>
      </Thead>
      <Tbody>
      <Tr>
        <Td>item 1</Td>
        <Td>item 2</Td>
      </Tr>
      </Tbody>
    </Table>
  )

  const headersA = ['alpha', 'beta']
  const headersB = ['one', 'two']

  // mount with initial headers
  const wrapper = mount(<Parent headers={headersA} i={1} />)
  expect(wrapper).toMatchSnapshot('headersA')

  // modify the headers by passing new prop, adjust key
  wrapper.setProps({ headers: headersB, i: 2 })
  expect(wrapper).toMatchSnapshot('headersB')
})

test('Renders colSpan', () => {
  let component = renderer.create(
    <Table>
      <Thead>
      <Tr>
        <Td>C1</Td>
        <Td>C2</Td>
        <Td>C3</Td>
      </Tr>
      </Thead>
      <Tbody>
      <Tr>
        <Td></Td>
        <Td colSpan="2">V4</Td>
        <Td>V6</Td>
      </Tr>
      </Tbody>
    </Table>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Renders rowSpan', () => {
  let component = renderer.create(
    <Table>
      <Thead>
      <Tr>
        <Td>C1</Td>
        <Td>C2</Td>
        <Td>C3</Td>
      </Tr>
      </Thead>
      <Tbody>
      <Tr>
        <Td>V3</Td>
        <Td rowSpan="2">V4</Td>
        <Td>V6</Td>
      </Tr>
      <Tr>
        <Td>V4</Td>
        <Td>V6</Td>
      </Tr>
      </Tbody>
    </Table>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Renders div in empty cell', () => {
  let component = renderer.create(
    <Table>
      <Thead>
      <Tr>
        <Td>C1</Td>
        <Td>C2</Td>
        <Td>C3</Td>
      </Tr>
      </Thead>
      <Tbody>
      <Tr>
        <Td></Td>
        <Td>V4</Td>
        <Td>V6</Td>
      </Tr>
      </Tbody>
    </Table>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

