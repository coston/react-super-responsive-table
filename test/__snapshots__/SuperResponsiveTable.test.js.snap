// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Table 1`] = `
<ContextProvider
  value={Object {}}
>
  <table
    className=" responsiveTable"
  >
    <Thead>
      <Tr>
        <Th>
          Annual Conference
        </Th>
        <Th>
          Year
        </Th>
        <Th>
          Location
        </Th>
        <Th>
          President
        </Th>
        <Th>
          Program Chair
        </Th>
        <Th>
          Conference Theme
        </Th>
      </Tr>
    </Thead>
    <Tbody>
      <Tr>
        <Td>
          31
        </Td>
        <Td>
          2017
        </Td>
        <Td>
          Alabama Community College System (ACCS)
        </Td>
        <Td>
          Mr. Toner Evans, Samford University
        </Td>
        <Td>
          Ms. Kelly Birchfield, Auburn University Montgomery
        </Td>
      </Tr>
    </Tbody>
  </table>
</ContextProvider>
`;

exports[`Render table with an only one column 1`] = `
<ContextProvider
  value={Object {}}
>
  <table
    className=" responsiveTable"
  >
    <Thead>
      <Tr>
        <Th>
          Annual Conference
        </Th>
      </Tr>
    </Thead>
    <Tbody>
      <Tr>
        <Td>
          31
        </Td>
      </Tr>
    </Tbody>
  </table>
</ContextProvider>
`;

exports[`Render table with conditional and unconditional column 1`] = `
<table
  className=" responsiveTable"
>
  <thead>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        C1
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        C3
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        V1
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        V3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Render table with conditional column 1`] = `
<table
  className=" responsiveTable"
>
  <thead>
    <tr />
  </thead>
  <tbody>
    <tr />
  </tbody>
</table>
`;

exports[`Render table with more columns in body 1`] = `
<table
  className=" responsiveTable"
>
  <thead>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        C1
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        C3
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        V1
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        V3
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        />
        V4
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        />
        V6
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Render table with more columns in header 1`] = `
<table
  className=" responsiveTable"
>
  <thead>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        C1
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        C3
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C4
        </div>
        C4
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C5
        </div>
        C5
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C6
        </div>
        C6
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        V1
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        V3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Render table with only header and empty row 1`] = `
<table
  className=" responsiveTable"
>
  <thead>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          Test Column
        </div>
        Test Column
      </td>
    </tr>
  </thead>
  <tbody>
    <tr />
  </tbody>
</table>
`;

exports[`Render table without any column 1`] = `
<ContextProvider
  value={Object {}}
>
  <table
    className=" responsiveTable"
  >
    <Thead>
      <Tr />
    </Thead>
    <Tbody>
      <Tr />
    </Tbody>
  </table>
</ContextProvider>
`;

exports[`Renders colSpan 1`] = `
<table
  className=" responsiveTable"
>
  <thead>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        C1
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C2
        </div>
        C2
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        C3
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        <div>
           
        </div>
      </td>
      <td
        colSpan="2"
      >
        V4
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        V6
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Renders div in empty cell 1`] = `
<table
  className=" responsiveTable"
>
  <thead>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        C1
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C2
        </div>
        C2
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        C3
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        <div>
           
        </div>
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C2
        </div>
        V4
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        V6
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Renders rowSpan 1`] = `
<table
  className=" responsiveTable"
>
  <thead>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        C1
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C2
        </div>
        C2
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        C3
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        V3
      </td>
      <td
        className=" pivoted"
        rowSpan="2"
      >
        <div
          className="tdBefore"
        >
          C2
        </div>
        V4
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C3
        </div>
        V6
      </td>
    </tr>
    <tr>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C1
        </div>
        V4
      </td>
      <td
        className=" pivoted"
      >
        <div
          className="tdBefore"
        >
          C2
        </div>
        V6
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table updates correctly upon change of key prop: headersA 1`] = `
<Parent
  headers={
    Array [
      "alpha",
      "beta",
    ]
  }
  i={1}
>
  <Table
    key="1"
  >
    <table
      className=" responsiveTable"
    >
      <Thead>
        <thead>
          <Tr
            inHeader={true}
          >
            <TrInner
              headers={
                Object {
                  "0": "alpha",
                  "1": "beta",
                }
              }
              inHeader={true}
            >
              <tr>
                <Th
                  columnKey={0}
                  key="0/.0"
                >
                  <th>
                    alpha
                  </th>
                </Th>
                <Th
                  columnKey={1}
                  key="1/.1"
                >
                  <th>
                    beta
                  </th>
                </Th>
              </tr>
            </TrInner>
          </Tr>
        </thead>
      </Thead>
      <Tbody>
        <tbody>
          <Tr>
            <TrInner
              headers={
                Object {
                  "0": "alpha",
                  "1": "beta",
                }
              }
            >
              <tr>
                <Td
                  columnKey={0}
                  key="0/.0"
                >
                  <TdInner
                    columnKey={0}
                    headers={
                      Object {
                        "0": "alpha",
                        "1": "beta",
                      }
                    }
                  >
                    <td
                      className=" pivoted"
                    >
                      <div
                        className="tdBefore"
                      >
                        alpha
                      </div>
                      item 1
                    </td>
                  </TdInner>
                </Td>
                <Td
                  columnKey={1}
                  key="1/.1"
                >
                  <TdInner
                    columnKey={1}
                    headers={
                      Object {
                        "0": "alpha",
                        "1": "beta",
                      }
                    }
                  >
                    <td
                      className=" pivoted"
                    >
                      <div
                        className="tdBefore"
                      >
                        beta
                      </div>
                      item 2
                    </td>
                  </TdInner>
                </Td>
              </tr>
            </TrInner>
          </Tr>
        </tbody>
      </Tbody>
    </table>
  </Table>
</Parent>
`;

exports[`Table updates correctly upon change of key prop: headersB 1`] = `
<Parent
  headers={
    Array [
      "one",
      "two",
    ]
  }
  i={2}
>
  <Table
    key="2"
  >
    <table
      className=" responsiveTable"
    >
      <Thead>
        <thead>
          <Tr
            inHeader={true}
          >
            <TrInner
              headers={
                Object {
                  "0": "one",
                  "1": "two",
                }
              }
              inHeader={true}
            >
              <tr>
                <Th
                  columnKey={0}
                  key="0/.0"
                >
                  <th>
                    one
                  </th>
                </Th>
                <Th
                  columnKey={1}
                  key="1/.1"
                >
                  <th>
                    two
                  </th>
                </Th>
              </tr>
            </TrInner>
          </Tr>
        </thead>
      </Thead>
      <Tbody>
        <tbody>
          <Tr>
            <TrInner
              headers={
                Object {
                  "0": "one",
                  "1": "two",
                }
              }
            >
              <tr>
                <Td
                  columnKey={0}
                  key="0/.0"
                >
                  <TdInner
                    columnKey={0}
                    headers={
                      Object {
                        "0": "one",
                        "1": "two",
                      }
                    }
                  >
                    <td
                      className=" pivoted"
                    >
                      <div
                        className="tdBefore"
                      >
                        one
                      </div>
                      item 1
                    </td>
                  </TdInner>
                </Td>
                <Td
                  columnKey={1}
                  key="1/.1"
                >
                  <TdInner
                    columnKey={1}
                    headers={
                      Object {
                        "0": "one",
                        "1": "two",
                      }
                    }
                  >
                    <td
                      className=" pivoted"
                    >
                      <div
                        className="tdBefore"
                      >
                        two
                      </div>
                      item 2
                    </td>
                  </TdInner>
                </Td>
              </tr>
            </TrInner>
          </Tr>
        </tbody>
      </Tbody>
    </table>
  </Table>
</Parent>
`;
